docs/payload_context.rst

```diff
@@ -0,0 +1,84 @@
+.. _payload_context:
+
+Payload Context
+===============
+
+Sometimes it is beneficial to pass some contextual information from the trigger
+to the trigger listener along the payload. Examples are:
+
+- tracing information that allows to track complex request processing in a
+  multi component system
+- in a multitenant system a tenant information to be able to identify the
+  tenant that peformed an operation that triggered a notification
+
+
+This can be done by using **Payload Context**. This feature includes:
+
+- ability to add an additional information to the payload in the trigger
+- ability to filter by the fields in the context in the listener process
+- ability to use ``context`` fields in the listener callbacks
+
+
+Add ``context`` to payload in the trigger
+-----------------------------------------
+
+Before doing updates that produce notifications set the context that should be
+passed using ``pgpubsub.set_notification_context`` function. 
+
+.. code-block:: python
+
+    from pgpubsub import set_notification_context
+
+    set_notification_context({'some-key': 'some-value'})
+
+The setting is effective till the connection is closed. Alternatively the setting
+``PGPUBSUB_TX_BOUND_NOTIFICATION_CONTEXT`` can be used to clean the context at the end 
+of the current transanction.
+
+
+Filter by ``context`` field in the trigger listener
+---------------------------------------------------
+
+Define a class that implements ``ListenerFilterProvider`` protocol and set
+option ``PGPUBSUB_LISTENER_FILTER`` to its fully qualified class name.
+
+.. code-block:: python
+
+    from pgpubsub import ListenerFilterProvider
+
+    class TenantListenerFilterProvider(ListenerFilterProvider):
+        def get_filter(self) -> Q:
+            return Q(payload__context__tenant='my-tenant')
```

Comment:
This will all only work for channels with `lock_notifications` right (i.e. with stored notifications)? Should we mention that somewhere?